lib/swoosh/adapters/mailgun.ex

```diff
@@ -66,8 +66,20 @@ defmodule Swoosh.Adapters.Mailgun do
     |> prepare_cc(email)
     |> prepare_bcc(email)
     |> prepare_reply_to(email)
+    |> prepare_custom_vars(email)
   end
 
+  # example custom_vars
+  # 
+  # %{"my_var" => Poison.encode!(%{"my_message_id": 123}), 
+  #   "my_other_var" => Poison.encode!(%{"my_other_id": 1, "stuff": 2})}
+  defp prepare_custom_vars(body, %Email{provider_options: custom_vars}) do
```

Comment:
We probably want to nest this under `custom_vars` or something similar. We may need to pass other provider-specific options that we don't want to end up as custom vars.